plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -49,23 +49,51 @@ import {
 import { TeamMember as AdoTeamMember } from 'azure-devops-node-api/interfaces/common/VSSInterfaces';
 import { Logger } from 'winston';
 import { PolicyEvaluationRecord } from 'azure-devops-node-api/interfaces/PolicyInterfaces';
-import { WebApi } from 'azure-devops-node-api';
+import { WebApi, getPersonalAccessTokenHandler } from 'azure-devops-node-api';
 import {
   TeamProjectReference,
   WebApiTeam,
 } from 'azure-devops-node-api/interfaces/CoreInterfaces';
 import { UrlReader } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  DefaultAzureDevOpsCredentialsProvider,
+  ScmIntegrations,
+} from '@backstage/integration';
 
 /** @public */
 export class AzureDevOpsApi {
   public constructor(
     private readonly logger: Logger,
-    private readonly webApi: WebApi,
     private readonly urlReader: UrlReader,
+    private readonly config: Config,
   ) {}
 
+  private async getWebApi(host?: string, org?: string): Promise<WebApi> {
+    const validHost = host ?? this.config.getString('azureDevOps.host');
+    const validOrg = org ?? this.config.getString('azureDevOps.organization');
+    const scmIntegrations = ScmIntegrations.fromConfig(this.config);
+    const credentialsProvider =
+      DefaultAzureDevOpsCredentialsProvider.fromIntegrations(scmIntegrations);
+    const credentials = await credentialsProvider.getCredentials({
+      url: `https://${validHost}/${validOrg}`,
+    });
+
+    let validToken: string;
+    if (credentials && credentials.token) {
+      validToken = credentials.token;
+    } else {
+      validToken = this.config.getString('azureDevOps.token');
+    }
+
+    const authHandler = getPersonalAccessTokenHandler(validToken);
```

Comment:
At this point, the `credential` could be a `bearer` or `pat` so passing the token to a `PersonalAccessTokenHandler` might fail depending on the integration config.

Reply:
Something like this should do the trick:
```suggestion
    const authHandler = this.config.getString('azureDevOps.token') || credentials?.type === 'pat'
          ? getPersonalAccessTokenHandler(credentials!.token ?? this.config.getString('azureDevOps.token') )
          : getBearerHandler(credentials!.token);
```